src/version.inc

```diff
@@ -0,0 +1,9 @@
+#define LC0_VERSION_MAJOR 0
+#define LC0_VERSION_MINOR 13
+#define LC0_VERSION_PATCH 0
+#define STR_INNER(a) #a
+#define STR(a) STR_INNER(a)
+#define LC0_VERSION_STRING     \
```

Comment:
## doesn't work between string constant and a numeric constant.

I moved the logic to the one place that needs it rather than have STR having to escape a header in order for LC0_VERSION_STRING to be useful.

I changed LC0_VERSION_PATCH to be 0 padded as requested, although I don't think its a great idea.

Reply:
Actually pushed the change to be 0 padded this time... heh.

Reply:
I don't like "0.01.02" because it's highly nonstandard, and is liable to confuse a lot of code that assumes the standard form.

As for the macros, `version.inc` should only contain the defines/constexprs whatever they are. How the version information is formatted into a string is unrelated to the actual version numbers themselves, and so version.inc should not have any string formatting code, or indeed any code at all as crem said.

(Data: https://en.wikipedia.org/wiki/Software_versioning#Schemes or https://semver.org/)